src/sentry/static/sentry/app/views/settings/projectAlerts/edit.tsx

```diff
@@ -24,36 +24,70 @@ type Props = RouteComponentProps<RouteParams, {}> & {
   hasMetricAlerts: boolean;
 };
 
-function ProjectAlertsEditor(props: Props) {
-  const {hasMetricAlerts, location, params, organization, project} = props;
-  const {projectId} = params;
-  const alertType = location.pathname.includes('/alerts/metric-rules/')
-    ? 'metric'
-    : 'issue';
-  const title = t('Edit Alert Rule');
-
-  return (
-    <React.Fragment>
-      <SentryDocumentTitle title={title} objSlug={projectId} />
-      <PageContent>
-        <BuilderBreadCrumbs
-          hasMetricAlerts={hasMetricAlerts}
-          orgSlug={organization.slug}
-          title={title}
-        />
-        <StyledPageHeader>
-          <PageHeading>{title}</PageHeading>
-        </StyledPageHeader>
-        {(!hasMetricAlerts || alertType === 'issue') && (
-          <IssueEditor {...props} project={project} />
-        )}
-
-        {hasMetricAlerts && alertType === 'metric' && (
-          <IncidentRulesDetails {...props} project={project} />
-        )}
-      </PageContent>
-    </React.Fragment>
-  );
+type State = {
+  alertType: string;
+  ruleName: string;
+};
+
+class ProjectAlertsEditor extends React.Component<Props, State> {
+  state: State = {
+    alertType: '',
+    ruleName: '',
+  };
+
+  handleChangeTitle = ruleName => {
+    this.setState({ruleName});
+  };
+
+  getTitle() {
+    const {ruleName} = this.state;
+    const defaultTitle = t('Edit Alert Rule');
+
+    const title = `${ruleName}`;
+
+    if (!ruleName) {
+      return defaultTitle;
+    }
```

Comment:
I would move this above where you declare and define `title, since `title` is only used if this condition is false.